util/stmtsummary/statement_summary.go

```diff
@@ -601,8 +601,26 @@ func (ssbd *stmtSummaryByDigest) collectHistorySummaries(historySize int) []*stm
 }
 
 func newStmtSummaryByDigestElement(sei *StmtExecInfo, beginTime int64, intervalSeconds int64) *stmtSummaryByDigestElement {
+	// sampleSQL / sampleUser / samplePlan / prevSQL / indexNames store the values shown at the first time,
+	// because it compacts performance to update every time.
+	maxSQLLength := config.GetGlobalConfig().StmtSummary.MaxSQLLength
+	sampleSQL := sei.OriginalSQL
+	if len(sampleSQL) > int(maxSQLLength) {
+		// Make sure the memory of original `sampleSQL` will be released.
+		sampleSQL = string([]byte(sampleSQL[:maxSQLLength]))
```

Comment:
Do we need to attach the information of sample query total length, e.g: `select * from tt (total: 2304bytes)`

Reply:
Addressed.